src/main/java/de/zalando/aruha/nakadi/controller/EventPublishingController.java

```diff
@@ -75,6 +81,17 @@ private ResponseEntity postEventInternal(String eventTypeName, String eventsAsSt
         }
     }
 
+    private ResponseEntity processJSONException(final JSONException e, final NativeWebRequest nativeWebRequest) {
+        if (e.getCause() == null) {
+            return create(crateProblem(e), nativeWebRequest);
+        }
+        return create(Problem.valueOf(Response.Status.BAD_REQUEST), nativeWebRequest);
+    }
+
+    private ThrowableProblem crateProblem(final JSONException e) {
```

Comment:
For me it looks cleaner because 'create' method is not overloaded with additional text.